web/handlers.go

```diff
@@ -1,3 +1,156 @@
 // Package web provides managing HTTP protocol.
 // Implement CRUD operations for filters over HTTP API.
 package web
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+	"github.com/lvl484/positioning-filter/repository"
+)
+
+const (
+	inputUserID = "user_id"
+	inputName   = "name"
+)
+
+type repo struct {
+	filters repository.Filters
+}
+
+func newRepo(filters repository.Filters) *repo {
+	return &repo{
+		filters: filters,
+	}
+}
+
+func (repo *repo) AddFilter(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	userIDstring := vars[inputUserID]
+
+	userid, err := uuid.Parse(userIDstring)
+	if err != nil {
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	var filter repository.Filter
+
+	if err := json.NewDecoder(r.Body).Decode(&filter); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	filter.UserID = userid
+
+	if err := repo.filters.Add(&filter); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusCreated)
+}
+
+func (repo *repo) GetOneFilter(w http.ResponseWriter, r *http.Request) {
+	m := mux.Vars(r)
+	filterName := m[inputName]
+	userIDstring := m[inputUserID]
+	userID, err := uuid.Parse(userIDstring)
+
+	if err != nil {
+		w.WriteHeader(http.StatusUnauthorized)
```

Comment:
Propose to respond NotFound in this case

Reply:
yep, thanks